httpbp/example_handler_test.go

```diff
@@ -0,0 +1,75 @@
+package httpbp_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/httpbp"
+	"github.com/reddit/baseplate.go/log"
+)
+
+type body struct {
+	X int `json:"x"`
+	Y int `json:"y"`
+}
+
+type errorResponse struct {
+	Reason  string `json:"reason"`
+	Message string `json:"message,omitempty"`
+}
+
+func home(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	response := httpbp.Response{
+		Body: body{
+			X: 1,
+			Y: 2,
+		},
+	}
+	return httpbp.WriteJSON(w, response)
+}
+
+func err(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	body := errorResponse{
+		Reason:  "BAD_REQUEST",
+		Message: "this endpoint always returns an error",
+	}
+	return httpbp.NewJSONError(http.StatusBadRequest, body, errors.New("example"))
+}
+
+func loggingMiddleware(next httpbp.HandlerFunc) httpbp.HandlerFunc {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		log.Infof("Request: %#v", r)
+		return next(ctx, w, r)
+	}
+}
+
+var (
+	_ httpbp.HandlerFunc = home
+	_ httpbp.Middleware  = loggingMiddleware
+)
+
+func ExampleBaseplateHandlerFactory() {
+	var (
+		ecImpl       *edgecontext.Impl
+		logger       log.Wrapper
+		trustHandler httpbp.HeaderTrustHandler
+	)
+	handlerFactory := httpbp.BaseplateHandlerFactory{
```

Comment:
🔕 I think eventually this function should be moved to top level and becomes `baseplate.HTTPHandlerFactory` instead. But definitely not happening in this PR.

Reply:
I don't think I agree since that means that _every_ service has to include the HTTP stuff as well.

Reply:
That's true, but isn't the direction https://github.com/reddit/baseplate.go/pull/133 is going means that eventually the top level will depend on both `thriftbp` and `httpbp` (so it provides battery-included helpers for both thrift and http implementations)?

Reply:
I was under the impression that the `baseplate` package would _explicitly_ only be things that are agnostic towards the type of service, I haven't checked in on that PR recently so I could be wrong.